routers/repo/reviews.go

```diff
@@ -0,0 +1,453 @@
+
+		commentUser, _ := models.GetUserByEmail(c.Comment.Author)
+		reviewComment.Poster = commentUser
+
+		reviewComment.Content = c.Comment.Description
+
+		reviewComment.RenderedContent = string(markdown.Render([]byte(reviewComment.Content), ctx.Repo.RepoLink,
+			ctx.Repo.Repository.ComposeMetas()))
+
+		commentCreatedTimestamp, _ := strconv.Atoi(reviewDetails.Request.Timestamp)
+		commentCreated := time.Unix(int64(commentCreatedTimestamp), 0)
+		reviewComment.Created = commentCreated
+
+		//
+		// 		// Check tag.
+		// 		tag, ok = marked[comment.PosterID]
+		// 		if ok {
+		// 			comment.ShowTag = tag
+		// 			continue
+		// 		}
+		//
+		if ctx.Repo.Repository.IsOwnedBy(reviewComment.Poster.ID) ||
+			(ctx.Repo.Repository.Owner.IsOrganization() && ctx.Repo.Repository.Owner.IsOwnedBy(reviewComment.Poster.ID)) {
+			reviewComment.ShowTag = models.CommentTagOwner
+		} else if reviewComment.Poster.IsWriterOfRepo(ctx.Repo.Repository) {
+			reviewComment.ShowTag = models.CommentTagWriter
+		} else if reviewComment.Poster.ID == issue.Poster.ID {
+			reviewComment.ShowTag = models.CommentTagPoster
+		}
+		//
+		// 		marked[comment.PosterID] = comment.ShowTag
+		//
+		isAdded := false
+		for j := range participants {
+			if reviewComment.Poster.ID == participants[j].ID {
+				isAdded = true
+				break
+			}
+		}
+		if !isAdded && !issue.IsPoster(reviewComment.Poster.ID) {
+			participants = append(participants, reviewComment.Poster)
+		}
+		// }
+		//
+		// pull := issue.PullRequest
+		// canDelete := false
+
+		// if ctx.IsSigned && pull.HeadBranch != "master" {
+		//
+		// 	if err := pull.GetHeadRepo(); err != nil {
+		// 		log.Error(4, "GetHeadRepo: %v", err)
+		// 	} else if ctx.User.IsWriterOfRepo(pull.HeadRepo) {
+		// 		canDelete = true
+		// 		deleteBranchURL := pull.HeadRepo.Link() + "/branches/" + pull.HeadBranch + "/delete"
+		// 		ctx.Data["DeleteBranchLink"] = fmt.Sprintf("%s?commit=%s&redirect_to=%s", deleteBranchURL, pull.MergedCommitID, ctx.Data["Link"])
+		//
+		// 	}
+		issue.Comments = append(issue.Comments, &reviewComment)
+	}
+
+	// ctx.Data["IsPullBranchDeletable"] = canDelete && git.IsBranchExist(pull.HeadRepo.RepoPath(), pull.HeadBranch)
+	//
+	ctx.Data["Participants"] = participants
+	ctx.Data["NumParticipants"] = len(participants)
+	ctx.Data["SignInLink"] = setting.AppSubURL + "/user/login?redirect_to=" + ctx.Data["Link"].(string)
+	ctx.HTML(200, tplIssueView)
+}
+
+// ViewReviewCommits show commits for a pull request
+func ViewReviewCommits(ctx *context.Context) {
+	ctx.Data["PageIsPullCommits"] = true
+
+	repo, err := repository.NewGitRepo(ctx.Repo.GitRepo.Path)
+	if err != nil {
+		ctx.Handle(500, "OpenGitRepository", err)
+	}
+
+	_, reviewDetails, _ := prepareReviewInfo(ctx)
+	if ctx.Written() {
+		return
+	}
+	ctx.Data["Username"] = ctx.Repo.Owner.Name
+	ctx.Data["Reponame"] = ctx.Repo.Repository.Name
+
+	reviewCommits, err := repo.ListCommitsBetween(reviewDetails.Request.TargetRef, reviewDetails.Request.ReviewRef)
```

Comment:
The logic for getting the commits in a review is a little more complicated than this because there are multiple states a review can be in (open, submitted, abandoned, etc).

We may actually want to add a 'ListCommits' method for reviews (in the git-appraise code) to encapsulate the logic, but in the meantime the best way to do this is something like:

    baseCommit, err := reviewDetails.GetBaseCommit()
    ...
    headCommit, err := reviewDetails.GetHeadCommit()
    ...
    reviewCommits, err := repo.ListCommitsBetween(baseCommit, headCommit)